packages/vite/rollup.config.ts

```diff
@@ -43,7 +43,17 @@ const clientConfig = defineConfig({
 
 const sharedNodeOptions = defineConfig({
   treeshake: {
-    moduleSideEffects: 'no-external',
+    moduleSideEffects(id, external) {
+      // These nested dependencies should be considered side-effect free
+      // as it's not set within their package.json
+      if (
+        id.includes('node_modules/astring') ||
+        id.includes('node_modules/acorn')
```

Comment:
I did knew this option but it didn't come up😅 Great that we can remove the patch.